src/server/helpers/achievement.js

```diff
@@ -401,7 +401,19 @@ export async function getEditsInDays(orm, editorId, days) {
 		and created_at > (SELECT CURRENT_DATE - INTERVAL '${days} days');`;
 
 	const out = await bookshelf.knex.raw(rawSql);
-	return out.rowCount;
+	const dateList = out.rows.map(row => row.created_at);
+	let countDays = 0;
+	let lastDate = new Date();
+	const oneDayMs = 24 * 60 * 60 * 1000;
```

Comment:
I can suggest a better way to compare dates (it gets very tricky; using 86400000 milliseconds as "a day" doesn't work for all cases.

We use the `date-fns` library in other parts of the codebase to remove the headache of dealing with time.

The `differenceInCalendarDays` utility should be what you are looking for https://date-fns.org/v2.29.2/docs/differenceInCalendarDays

When you map your rows to dateList, if created_at is a string rather than a Date object you'll need to transform it with `parseISO` from date-fns.